### PR TITLE
Support for Service

### LINKED ISSUE #22: Support creating, listing, getting, and deleting services
Services are just another kind of object.  Do the straightforward thing.

### PR DESCRIPTION
**Pull request by exarkun**
Fixes #22

### CODE REVIEW

File: src/txkube/_memory.py [hunk 1]
```diff
@@ -104,6 +105,15 @@ def response(request, status, obj):
 
 
 
+def _message(details, event):
+    if details.get(u"group") is None:
+        fmt = u"{kind} \"{name}\" {event}"
```
Comment (markrwilliams) [inline src/txkube/_memory.py]: why not use ' instead of "?  then you wouldn't need to escape the quotes.

File: src/txkube/testing/integration.py [hunk 2]
```diff
@@ -102,14 +106,352 @@ def _named(kind, name, namespace=None):
...
+        """
+        return self._create_duplicate_rejected_test(
+            None, creatable_namespaces(), u"namespaces", None,
+        )
+
+
+
+class _ConfigMapTestsMixin(TestCase):
+    @async
+    @needs(namespace=creatable_namespaces().example())
```
Comment (markrwilliams) [inline src/txkube/testing/integration.py:251]: it looks like everywhere you use `@needs` in this diff you use a hypothesis strategy.   also, hypothesis doesn't save or minimize `.example()`s.  

should `@needs` integrate more tightly with hypothesis?  with some wrangling you could give it a `data` strategy instance which would let you use `data.draw`: http://hypothesis.readthedocs.io/en/latest/data.html?highlight=draw#drawing-interactively-in-tests.

but you probably just want to make it use strategies directly.

**Review by markrwilliams — COMMENTED** [review]
Minor comments.  The approach looks good.

File: src/txkube/_memory.py (hunk 1 above)
Comment (exarkun) [reply src/txkube/_memory.py]: Yea that makes sense ... double quoting everything is mostly just a habit.  Cases where single quotes actually improve readability don't come up very often, didn't occur to me here. :)

File: src/txkube/testing/integration.py (hunk 2 above)
Comment (exarkun) [reply src/txkube/testing/integration.py:251]: Oohh.  `data.draw`.  That seems to be the kind of thing I was looking for when I was first writing `needs`.  Yes, I think I will do something with that.  I'm filing a ticket for it now.

**Pull request merged**